src/Symfony/Component/Security/Http/Firewall/AccessListener.php

```diff
@@ -68,7 +68,14 @@ public function __invoke(RequestEvent $event)
             $this->tokenStorage->setToken($token);
         }
 
-        if (!$this->accessDecisionManager->decide($token, $attributes, $request)) {
+        $granted = false;
+        foreach ($attributes as $key => $value) {
+            if ($this->accessDecisionManager->decide($token, [$key => $value], $request)) {
+                $granted = true;
```

Comment:
You could break the loop already here too.

Reply:
Good catch, fixed in https://github.com/symfony/symfony/commit/040d93d5b563052cfdeec0cade83aae86672bdfb